projects/js-packages/publicize-components/src/components/post-publish-share-status/index.tsx

```diff
@@ -33,13 +33,25 @@ export function PostPublishShareStatus() {
 
 	const willPostBeShared = isPublicizeEnabled && enabledConnections.length > 0;
 
-	if ( ! featureFlags.useShareStatus || ! willPostBeShared || ! isPostPublised ) {
+	const showStatus = featureFlags.useShareStatus && willPostBeShared && isPostPublised;
+
+	usePostJustPublished( () => {
+		if ( showStatus ) {
+			pollForPostShareStatus( {
+				isRequestComplete( { postShareStatus } ) {
+					return postShareStatus.done;
+				},
+			} );
+		}
+	}, [ showStatus ] );
+
+	if ( ! showStatus ) {
```

Comment:
Should this return be above the `usePostJustPublished`?

Reply:
No, [React hooks cannot rendered/called conditionally](https://react.dev/warnings/invalid-hook-call-warning).